<!DOCTYPE html>
<head>
    <link rel="SHORTCUT ICON" href="./logo.ico"/>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>TSVdb</title>
    <meta name="author" content="Sun Wenjie, Ye Panmeng" />
    <meta name="description" content="TCGA splicing variants database." />
    <meta name="keywords"  content="TCGA,tumor,splicing variants ,visualization" />
    <meta name="Resource-type" content="Document" />
    <link rel="stylesheet" type="text/css" href="./css/jquery.fullPage.css" />
    <link rel="stylesheet" type="text/css" href="./css/index.css"/>
    <link rel="stylesheet" type="text/css" href="./fonts/iconfont.css">
    <!-- Global Site Tag (gtag.js) - Google Analytics -->
    <script async src="https://www.googletagmanager.com/gtag/js?id=UA-106948170-1"></script>
    <script>
        window.dataLayer = window.dataLayer || [];
        function gtag(){dataLayer.push(arguments)};
        gtag('js', new Date());

        gtag('config', 'UA-106948170-1');
    </script>
    <script type="text/javascript" src="./js/sv/check_browser.js"></script>
    <script src="./js/jquery-3.2.1.min.js"></script>
    <script src="./js/jquery-ui.min.js"></script>
    <script type="text/javascript" src="./js/scrolloverflow.min.js"></script>
    <script type="text/javascript" src="./js/jquery.fullPage.min.js"></script>
    <!--    <script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=AM_HTMLorMML"></script>-->
    <script type="text/javascript">
        $(document).ready(function() {
            $('#fullpage').fullpage({
                //                sectionsColor: ['#81d4fa', '#e1f5fe', '#b3e5fc', ],
                sectionsColor: ['#e1f5fe', '#b3e5fc', '#fff'],
                navigationTooltips: ['Instruction', 'Tutorial', 'Toolchart'],
                navigation: true,
                css3: true,
                loopBottom: true,
                controlArrowColor: '#fff'
            });
        });
    </script>
</head>
<body>
    <div class="toHome">
        <a href="./index.html"><i class="iconfont icon-gene"></i>TSVdb</a>
    </div>
    <div class="toPlot">
        <a href="./plot.html">Quick Start -></a>
    </div>
    <div id="fullpage">

        <div class="section active" id="section_intro">
            <h1 class="intr_h1">TSVdb</h1>
            <h2 class="intr_h2">TCGA Splicing Variants DB</h2>
            <div class="intro"><p>TSVdb is an easy-to-use web tool for the integration and visualization of mRNA alternative splicing, transcriptional isoform expression and clinical information from The Cancer Genome Atlas project (TCGA) RNA-Seq data.</div>
            <!-- <div class="selection" id="selection_start"><a href="./plot_demo.html">Start</a></div> -->
            <!-- <div class="selection" id="selection_tutorial">Tutorial</div> -->
            <!--            <img src = "./image/strategy.svg" alt="strategy" style="height: 250px;" class="myImg">-->
            <img src = "./image/amination.gif" alt="strategy" style="height: 300px;" class="gif">
        </div>

        <div class="section" id="section_tutorial">
            <div class="slide" id="selecting_data">
                <h1 class="spe_h1">Tutorial</h1>
                <h2 class="sep_h2">1. Selecting Data</h2>
                <div class="slide_content">
                    <img src="./image/select_data.png" alt="select_data" style="height: 300px">
                    <p style="height: 0px"></p>
                    <p>
                    When open the data visualizing page, the user need to sequentially choose the tumor type, gene, data type and clinical information. Before completing the option choosing, no other operation can be performed. Do not afraid when making a mistake at this stage, the options can be edited afterward.
                    </p>
                    <p>
                    The first pop-up dialog (Fig. A) is cancer type, please select the cancer type you are interested in and click the selection button. In the second pop-up dialog (fig B), please enter an Official Gene Symbol or Entrez ID. If the query does not exist, the alarm would appear to notify you. Then choose the exon/junction usage value to be displayed (Fig. C). The exon/junction usage is the scaled expression, more about the scaling method, please refer to the next section. At last, choose the phenotype you are interesting and the gene expression data used to sort sample inside each phenotype subgorup (Fig. D). The gene expression value can be set to another gene so that the correlation between alternative splicing of one gene and the expression of another gene can be investigated.             
                    </p>
                </div>
            </div>

            <div class="slide" id="exon_junc">
                <h1 class="spe_h1">Tutorial</h1>
                <h2 class="sep_h2">2. About graph</h2>
                <div class="slide_content">
                    <img src="./image/about_graph.png" alt="about_graph" style="height: 400px">
                    <p>
                    In the header displays the options, they can be edited by clicking the corresponding buttons in the header.
                    </p>
                    <p>
                    The samples are firstly sorted by clinical type, then are sorted by the gene expression. When the mouse hovered the clinical attribution row, the group name and a red line will appear.
                    </p>
                    <p>
                    The “Exon or Junction” grey boxes are the exon/junction that TCGA used to count the mapping reads, in the TCGA Level 3 mRNA Seq datasets.
                    </p>
                    <p>
                    Each exon/junction usage value displays as shadowed line graph by row with samples by column. Exon/junction pattern will be transiently or permanently highlighted when the mouse hovered or clicked an exon/junction shadowed line graph. What’s more, when you double clicked on the exon/junction shawdowed line the UCSC Genome Browser will open in a new tab/window with selected exon/junction area highlighted and let you enjoy the annotation in UCSC Genome Browser.                     
                    </p>
                </div>
            </div>

            <div class="slide" id="isoform">
                <h1 class="spe_h1">Tutorial</h1>
                <h2 class="sep_h2">3. Isoform expression</h2>
                <div class="slide_content">
                    <img src="./image/about_isoform.png" alt="about_isoform" style="height: 300px">
                    <p>
                    The box plot of isoform expression (normalized RSEM) will appear after clicked the transcriptional isoform graph. At the top right side, you can see the UCSC isoform id, click on which will open a new tab or window display the UCSC Genome Browsers with the isoform highlighted. At the top of the pop-up box, the buttons are used to choose box plot or KM survival plot to be displayed. 
                    </p>
                    <p>
                    The KM survival curve included four parts: 1). In the bottom right is the part which is used to adjust the cut-point for grouping, in which the knob can be pulled to change the cutpoint. Moreover, the default value is the middle of the range. 2). The top right part displays information of groups include the cut-point value and sample size of each group. 3). The top left part shows the survival line of each person that enters after filtering by the survival start time point.  
                    </p>
                </div>
            </div>

            <div class="slide" id="data_table">
                <h1 class="spe_h1">Tutorial</h1>
                <h2 class="sep_h2">4. Download data table</h2>
                <div class="slide_content">
                    <table class="table", style="border-style: solid">
                        <thead>
                            <tr>
                                <th style="text-align: left">Attribute</th>
                                <th>Example</th>
                                <th>Description</th>
                            </tr>
                        </thead>
                        <tbody style="text-align: left; border-style: solid">
                            <tr>
                                <td style="text-align: left">sampleID</td>
                                <td>TCGA-3L-AA1B-01A</td>
                                <td>TCGA sample id.</td>
                            </tr>
                            <tr>
                                <td style="text-align: left">clinical_[clinical type]</td>
                                <td>Primary Solid Tumor</td>
                                <td>The phenotype information about the sample.</td>
                            </tr>
                            <tr>
                                <td style="text-align: left">os_time</td>
                                <td>348</td>
                                <td>Overall survival time (day).</td>
                            </tr>
                            <tr>
                                <td style="text-align: left">os_event</td>
                                <td>0</td>
                                <td>Overall survival event (0=live/censored, 1=death).</td>
                            </tr>
                            <tr>
                                <td style="text-align: left">gene_[gene name]</td>
                                <td>100</td>
                                <td>Gene expression normalized read count.</td>
                            </tr>
                            <tr>
                                <td style="text-align: left">isoform_[isoform id]</td>
                                <td>100</td>
                                <td>Isoform expression normalized read count.</td>
                            </tr>
                            <tr>
                                <td style="text-align: left">exon_[location]</td>
                                <td>0.1102</td>
                                <td>Exon/junction usage value. Exon/junction usage is calculated by dividing exon/junction quantification value</br>
                                    to mean exon/junction quantification value of that person for a specific gene</td>
                            </tr>
                        </tbody>
                    </table>
                </div>
            </div>
        </div>

        <div class="section" id="toolchart">
            <h1 class="spe_h1">Tool-chains and algorithms</h1>
            <div class="slide_content">
                <img src="./image/flowchart.png" alt="toolchart" style="height: 450px; float: left; margin: 0 30px 0 0">
                <div style="height: 50px"></div>
                <p>
                The TCGA level 3 gene, isoform, exon and junction normalized read count data (UNC illuminaHiSeq_RNASeqV2), clinical data (Merged_clinical_level_1) were downloaded from TCGA ftp site -- Firehose.
                </p>
                <p>
                The R was used to generate transcriptional pattern and TCGA data cleanup. The formatted JSON data were import into MongoDB.
                </p>
                <p>
                The exon/junction usage value was derived from exon/jucntion quantification. Firstly, each sample's exon/junction quantification value is divided by the expression quantity of the gene  to which exon belongs. So the gene expression effect was removed, and the AS event was highlighted. Next, the d3.js linear scale was used to map the normalized exon/junction value into the graph coordination. The interval of the exon/junction usage value (domain argument in scale function) is set to 0 - Q95 (95\% quantile) to diminish the outlier's impact. Furthermore, if the value of Q95 is smaller than 0.05, the upper bound of the interval would be set to 0.05.
            </div>
        </div>
    </div>
</body>
