<!DOCTYPE html>
<head>
    <link rel="SHORTCUT ICON" href="./logo.ico"/>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>TSVdb</title>
    <meta name="author" content="Sun Wenjie" />
    <meta name="description" content="TCGA splicing variants database." />
    <meta name="keywords"  content="TCGA,tumor,splicing variants ,visualization" />
    <meta name="Resource-type" content="Document" />
    <link rel="stylesheet" type="text/css" href="./css/jquery.fullPage.css" />
    <link rel="stylesheet" type="text/css" href="./css/index.css"/>
    <link rel="stylesheet" type="text/css" href="./fonts/iconfont.css">
    <!-- Global Site Tag (gtag.js) - Google Analytics -->
    <script async src="https://www.googletagmanager.com/gtag/js?id=UA-106948170-1"></script>
    <script>
        window.dataLayer = window.dataLayer || [];
        function gtag(){dataLayer.push(arguments)};
        gtag('js', new Date());

        gtag('config', 'UA-106948170-1');
    </script>
    <script type="text/javascript" src="./js/sv/check_browser.js"></script>
    <script src="./js/jquery-3.2.1.min.js"></script>
    <script src="./js/jquery-ui.min.js"></script>
    <script type="text/javascript" src="./js/scrolloverflow.min.js"></script>
    <script type="text/javascript" src="./js/jquery.fullPage.min.js"></script>
    <!--    <script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=AM_HTMLorMML"></script>-->
    <script type="text/javascript">
        $(document).ready(function() {
            $('#fullpage').fullpage({
                //                sectionsColor: ['#81d4fa', '#e1f5fe', '#b3e5fc', ],
                sectionsColor: ['#e1f5fe', '#b3e5fc', '#fff'],
                navigationTooltips: ['Instruction', 'Tutorial', 'Toolchart'],
                navigation: true,
                css3: true,
                loopBottom: true,
                controlArrowColor: '#fff'
            });
        });
    </script>
</head>
<body>
    <div class="toHome">
        <a href="./index.html"><i class="iconfont icon-gene"></i>TSVdb</a>
    </div>
    <div class="toPlot">
        <a href="./plot.html">Quick Start -></a>
    </div>
    <div id="fullpage">

        <div class="section active" id="section_intro">
            <h1 class="intr_h1">TSVdb</h1>
            <h2 class="intr_h2">TCGA Splicing Variants DB</h2>
            <div class="intro"><p>TSVdb is an easy-to-use web tool for the integration and visualization of mRNA splicing variants, transcriptional isoform expression and clinical information from The Cancer Genome Atlas project (TCGA) RNASeq data.</p></div>
            <!-- <div class="selection" id="selection_start"><a href="./plot_demo.html">Start</a></div> -->
            <!-- <div class="selection" id="selection_tutorial">Tutorial</div> -->
            <!--            <img src = "./image/strategy.svg" alt="strategy" style="height: 250px;" class="myImg">-->
            <img src = "./image/amination.gif" alt="strategy" style="height: 300px;" class="gif">
        </div>

        <div class="section" id="section_tutorial">
            <div class="slide" id="selecting_data">
                <h1 class="spe_h1">Tutorial</h1>
                <h2 class="sep_h2">1. Selecting Data</h2>
                <div class="slide_content">
                    <img src="./image/select_data.png" alt="select_data" style="height: 300px">
                    <p style="height: 0px"></p>
                    <p>
                    When open the data visualizing page, the user must sequentially select the tumor type, gene, data type and clinical information.
                    Before completing the selection, no other operation can be performed.
                    Do not afraid when making a mistake in this stage, the options can be edited afterward.
                    </p>
                    <p>
                    The first popup menu (fig A) is on cancer type, please choose the cancer type you are interested in and click the selection button.
                    In the second popup menu (fig B), please enter an Official Gene Symbol or Entrez ID. If the query term does not exist in the database, the alarm would appear to notify you. 
                    Then choose the Exon/Junction quantification usage value to be displayed (fig C). More about the scaling method please refer to the next section
                    At last, choose the phenotype you are interesting (fig D). The gene expression value can be set to another gene, so that the correlation between the splicing vairants and the gene can be investigated. 
                    </p>
                </div>
            </div>

            <div class="slide" id="exon_junc">
                <h1 class="spe_h1">Tutorial</h1>
                <h2 class="sep_h2">2. About graph</h2>
                <div class="slide_content">
                    <img src="./image/about_graph.png" alt="about_graph" style="height: 400px">
                    <p>
                    Options can be adjusted by clicking the corresponding buttons in the header.
                    </p>
                    <p>
                    The samples were sorted by clinical type, then sorted by the gene expression. When the mouse hovers the clinical attribution row, the group name and a guide line will appear.
                    </p>
                    <p>
                    The Exon or Junction pattern is the locations of exon/junction that TCGA used to count the mapping reads. They represent the information that can be drawn from the Level 3 data of the mRNA Seq datasets from TCGA.
                    </p>
                    <p>
                    Each exon/junction usage value was display as area graph by row with samples by column. A transient highlight of exon or junction pattern generated when the mouse hovers an exon or junction area graph. A durable highlight can be achieved by a single click. What more, the double click of an exon or junction area graph will open the UCSC Genome Browsers in a new tab or windows with selecting exon or junction area highlighted.
                    </p>
                </div>
            </div>

            <div class="slide" id="isoform">
                <h1 class="spe_h1">Tutorial</h1>
                <h2 class="sep_h2">3. Isoform expression</h2>
                <div class="slide_content">
                    <img src="./image/about_isoform.png" alt="about_isoform" style="height: 300px">
                    <p>
                    By clicking the transcriptional isoform, the box plot of isoform expression value (normalized RSEM) appears. At the top of the pop-up box, there are buttons used to switch the display of the box plot or the KM survival curve. At the top right side, you can see the UCSC isoform id, Click on the text will open a new tab or window display the UCSC Genome Browsers with the isoform highlighted.
                    </p>
                    <p>
                    The KM survival curve includes four parts: the bottom right part is the part used to adjust the cut-point for grouping; the knob can be pulled to change the cutpoint. The default value is the middle of the range. The top right part displays the grouping information includes the cut-point value and sample size of each group. The top left part shows the survival line of each person that entered after filtering by the survival start time point.
                    </p>
                </div>
            </div>

            <div class="slide" id="data_table">
                <h1 class="spe_h1">Tutorial</h1>
                <h2 class="sep_h2">4. Download data table</h2>
                <div class="slide_content">
                    <table class="table", style="border-style: solid">
                        <thead>
                            <tr>
                                <th style="text-align: left">Attribute</th>
                                <th>Example</th>
                                <th>Description</th>
                            </tr>
                        </thead>
                        <tbody style="text-align: left; border-style: solid">
                            <tr>
                                <td style="text-align: left">sampleID</td>
                                <td>TCGA-3L-AA1B-01A</td>
                                <td>TCGA sample id.</td>
                            </tr>
                            <tr>
                                <td style="text-align: left">clinical_[clinical type]</td>
                                <td>Primary Solid Tumor</td>
                                <td>Estimate for the absolute depth of read coverage across the whole transcript</td>
                            </tr>
                            <tr>
                                <td style="text-align: left">os_time</td>
                                <td>348</td>
                                <td>Overall survival time (day).</td>
                            </tr>
                            <tr>
                                <td style="text-align: left">os_event</td>
                                <td>0</td>
                                <td>Overall survival event (0=live, 1=death).</td>
                            </tr>
                            <tr>
                                <td style="text-align: left">gene_[gene name]</td>
                                <td>100</td>
                                <td>Gene expression RSEM value.</td>
                            </tr>
                            <tr>
                                <td style="text-align: left">isoform_[isoform id]</td>
                                <td>100</td>
                                <td>Isoform expression RSEM value.</td>
                            </tr>
                            <tr>
                                <td style="text-align: left">exon_[location]</td>
                                <td>0.1102</td>
                                <td>Exon/junction usage value. Exon/junction usage is calculated by dividing exon/junction quantification value</br>
                                    to mean exon/junction quantification value of that person for a specific gene</td>
                            </tr>
                        </tbody>
                    </table>
                </div>
            </div>
        </div>

        <div class="section" id="toolchart">
            <h1 class="spe_h1">Tool-chains and algorithms</h1>
            <div class="slide_content">
                <img src="./image/flowchart.png" alt="toolchart" style="height: 450px; float: left; margin: 0 30px 0 0">
                <div style="height: 50px"></div>
                <p>
                The TCGA level 3 gene and isoform RSEM expression, exon and junction quantification value (UNC illuminaHiSeq_RNASeqV2), clinical data (Merged_clinical_level_1) were downloaded from TCGA ftp site -- Firehouse.
                </p>
                <p>
                The R was used to generate transcriptional pattern and TCGA data cleanup. The formatted JSON data were import into MongoDB.
                </p>
                <p>
                The exon/junction usage value was derived from exon/jucntion quantification. Firstly, each sample's exon/junction quantification value is divided by the expression quantity of the gene  to which exon belongs. So the gene expression effect was removed, and the AS event was highlighted. Next, the d3.js linear scale was used to map the normalized exon/junction value into the graph coordination. The interval of the exon/junction usage value (domain argument in scale function) is set to 0 - Q95 (95\% quantile) to diminish the outlier's impact. Furthermore, if the value of Q95 is smaller than 0.05, the upper bound of the interval would be set to 0.05.
            </div>
        </div>
    </div>
</body>
</html>
